crates/subspace-networking/src/behavior/custom_record_store.rs

```diff
@@ -260,35 +262,39 @@ impl<'a> RecordStorage<'a> for NoRecordStorage {
     }
 }
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
+use parity_scale_codec::{Decode, Encode};
+
+#[derive(Clone, Debug, Serialize, Deserialize, Decode, Encode)]
 struct ParityDbRecord {
-    /// Key of the record.
-    pub key: Key,
-    /// Value of the record.
-    pub value: Vec<u8>,
-    /// The (original) publisher of the record.
-    pub publisher: Option<PeerId>,
+    // Key of the record.
+    key: Vec<u8>,
+    // Value of the record.
+    value: Vec<u8>,
+    // The (original) publisher of the record.
+    publisher: Option<Vec<u8>>,
     // TODO: add expiration field and convert Instant to serializable time-type
-    // /// The expiration time as measured by a local, monotonic clock.
-    // pub expires: Option<Instant>,
+    // // The expiration time as measured by a local, monotonic clock.
+    // expires: Option<Instant>,
 }
 
 impl From<Record> for ParityDbRecord {
     fn from(rec: Record) -> Self {
         Self {
-            key: rec.key,
+            key: rec.key.to_vec(),
             value: rec.value,
-            publisher: rec.publisher,
+            publisher: rec.publisher.map(|peer_id| peer_id.to_bytes()),
         }
     }
 }
 
 impl From<ParityDbRecord> for Record {
     fn from(rec: ParityDbRecord) -> Self {
         Self {
-            key: rec.key,
+            key: rec.key.into(),
             value: rec.value,
-            publisher: rec.publisher,
+            publisher: rec
+                .publisher
+                .map(|peer_id| PeerId::from_bytes(&peer_id).expect("Fatal serialization error.")),
```

Comment:
Well, this is de-serialization and `expect()` should convince the reader of the code why this will never happen or why it is okay to crash the whole app if it does happen. "Fatal serialization error." doesn't help much with that.